libraries/AP_HAL/SPIDevice.h

```diff
@@ -43,8 +43,12 @@ class SPIDevice : public Device {
     virtual Semaphore *get_semaphore() override = 0;
 
     /* See Device::register_periodic_callback() */
-    virtual Device::PeriodicHandle *register_periodic_callback(
-        uint32_t period_usec, MemberProc) override = 0;
+    virtual Device::PeriodicHandle register_periodic_callback(
+        uint32_t period_usec, Device::PeriodicCb) override = 0;
+
+    /* See Device::adjust_periodic_callback() */
+    virtual bool adjust_periodic_callback(
+        PeriodicHandle h, uint32_t period_usec) override { return false; }
```

Comment:
I didn't want  to break the build or having to add it on PX4... we can remove this once it's implemented everywhere.